### ISSUE TITLE
Question: how to add image class to model properly?

### ISSUE THREAD

**Issue by mudkip-hacker** [WongKinYiu/yolor#96]
This is my steps so far:

1. add new .cfg file, changing all `classes=80` to `classes=81`, changing all `filters=255` to `filters=258`
2. add new yaml file, changing `nc: 80` to `nc:81`, and adding a name to `names: []` (i.e., empty shelf)
3. add new .names file with the added name (i.e., empty shelf)
4. download the pretrained `yolor_w6.pt`
5. train the model with my custom data, using `python train.py --batch-size 4 --img 1280 1280 --data my_yaml.yaml --cfg cfg/my_config.cfg --weights yolor_w6.pt --device 0 --name my_run --hyp hyp.scratch.1280.yaml --epochs 30`
6. inference with `python detect.py --source inference/images/ --cfg cfg/my_config.cfg --weights runs/train/my_run/weights/last_029.pt --conf 0.2 --img-size 1280 --device 0 --names data/my_namesl.names`

The output does indeed show my image classes being detected, but all the other ones are gone? I couldn't find any resources for proper steps for extending the model, so any help/correction with the above steps is appreciated

example non extended model inference:
![image](https://user-images.githubusercontent.com/34434655/133497858-0c446356-42a9-4d22-b500-25bd3f82a0f1.png)

example attempted extended model inference, with desired "refrigerator" class missing:
![image](https://user-images.githubusercontent.com/34434655/133497940-3212b962-42db-40c9-8907-1ad483adf025.png)

**Comment by hiteshhedwig** [comment]
Did you figure out anything?

**Comment by mudkip-hacker** [comment]
> Did you figure out anything?

No, but I guess I have what I needed lol

**Comment by WongKinYiu** [comment]
because the shape of weights of the detection layers are different, they will be re-initialize by default.

in this case, to simplify the problem,  i suggest you to append pseudo label of coco classes to your own label in your custom dataset.

(the better way is copy weights of original 80 classes to new weights, and do transfer/incremental learning on the new class.)
